### ISSUE TITLE
Missing root domain from `link('name').address`

### ISSUE THREAD

**Issue by dpb587** [cloudfoundry/bosh#1822]
When running bosh/263.4.0, in a deployment with a single VM that provides+consumes a link, and referring to that link's address in a template...

```
        <% if_link('webapp') do |link| %>
            set $upstream_webapp http://<%= link.address %>:<%= link.p('port') %>;
```

I see the following rendered content which excludes the default, non-overridden `bosh` root domain...

```
            set $upstream_webapp http://q-s0.main.private.legacy-develop.:8080;
```

I did not look into why, but changing `root_domain` to `domain` in the following file...

https://github.com/cloudfoundry/bosh/blob/c41ffef659b4a490914232db66385ff2c7a40c3a/src/bosh-template/lib/bosh/template/evaluation_context.rb#L152

It happens to render correctly with the following...

```
            set $upstream_webapp http://q-s0.main.private.legacy-develop.bosh:8080;
```

I didn't look into why.

**Comment by dpb587-pivotal** [comment]
Story [#152443729](https://www.pivotaltracker.com/story/show/152443729)

**Comment by dpb587-pivotal** [comment]
Closing - fixed in v264.1.
